docs/source/services.md

```diff
@@ -0,0 +1,84 @@
+# JupyterHub services
+
+JupyterHub 0.7 adds the notion of Services.
+A Service is a process that interacts with the Hub REST API.
+Services may perform actions such as shutting down user servers that have been idle for some time,
+or registering additional web servers that should also use the Hub's authentication
+and be served behind the Hub's proxy.
+
+There are two main characteristics of services:
+
+1. Is it **managed** by JupyterHub?
+2. Does it have a web server that should be added to the proxy?
+
+If a `command` is specified for launching the service, it will be started and managed by the Hub.
+If a `url` is specified for where the service runs its own webserver,
+it will be added to the Hub's proxy at `/service/:service-name`.
+
+## Managed services
+
+**Managed** services are services that the Hub starts and is responsible for.
+These can only be local subprocesses of the Hub.
+While there are similarities with notebook Spawners,
+there are no plans to support the same spawning abstractions as notebook
+If you want to run these services in docker or other environments,
```

Comment:
>  do you think there's a benefit of having configurable extension points vs telling advanced users to subclass and override JupyterHub.initialize and JupyterHub.cleanup?

Not really, and subclass makes it a bit clearer who the appropriate audience for such customization is, which I like. I should document that, though, so it's not quite at the "if you know where to look" level.

>  If that were the API though, seems a little odd to also support passing dictionaries, since those are presumably getting turned into Service objects by the hub anyway.

I agree that it's weird. I wanted to do something a bit more declarative so there are fewer black boxes to deal with, and the capabilities are explicitly not another arbitrary scope-expansion point, but if I end up allowing an instance, I would probably make it *always* an instance as you suggest, instead of sometimes a dict and sometimes an instance.

A service is, by definition here, something that has API access to the Hub, so it will go in the DB and get an API token. That might be one point against using services as the extension point for things that don't actually want that.